kratos/utilities/exact_mortar_segmentation_utility.cpp

```diff
@@ -505,18 +507,18 @@ bool ExactMortarIntegrationUtility<2, 2, true>::GetExactIntegration(
     const double tolerance = 1.0e3 * ZeroTolerance;
 
     double total_weight = 0.0;
-    array_1d<double, 2> auxiliar_coordinates(2, 0.0);
+    array_1d<double, 2> auxiliar_coordinates = ZeroVector(2);
     array_1d<PointBelongsLine2D2N, 2> auxiliar_belong;
 
     // Declaring auxiliar values
     PointType projected_gp_global;
-    GeometryNodeType::CoordinatesArrayType projected_gp_local;
+    GeometryType::CoordinatesArrayType projected_gp_local;
 
     // First look if the edges of the slave are inside of the master, if not check if the opposite is true, if not then the element is not in contact
     for (unsigned int i_slave = 0; i_slave < 2; ++i_slave) {
-        const array_1d<double, 3>& normal = rOriginalSlaveGeometry[i_slave].FastGetSolutionStepValue(NORMAL);
```

Comment:
can this be refactored to removed the necessity for computing the normals?
This would make the use for mapping much much easier

Reply:
I can, but It will break the Contact app right now, I can do it in an independent PR

Reply:
The NORMAL is assumed to be computed as average an then smoothed BTW

Reply:
ok no problem, was more a question for the longer term, no hurries

Maybe you can use the function that I introduced recently for projecting on a line